AK/Vector.h

```diff
@@ -275,22 +277,38 @@ class Vector {
             }
         }
         new (slot(index)) T(forward<U>(value));
+        return true;
+    }
+
+    template<typename U = T>
+    void insert(size_t index, U&& value)
+    {
+        VERIFY(try_insert<U>(index, forward<U&&>(value)));
     }
 
     template<typename C, typename U = T>
-    void insert_before_matching(U&& value, C callback, size_t first_index = 0, size_t* inserted_index = nullptr)
+    [[nodiscard]] bool try_insert_before_matching(U&& value, C callback, size_t first_index = 0, size_t* inserted_index = nullptr)
     {
         for (size_t i = first_index; i < size(); ++i) {
             if (callback(at(i))) {
-                insert(i, forward<U>(value));
+                if (!try_insert(i, forward<U>(value)))
+                    return false;
                 if (inserted_index)
                     *inserted_index = i;
-                return;
+                return true;
             }
         }
-        append(forward<U>(value));
+        if (!try_append(forward<U>(value)))
+            return false;
         if (inserted_index)
             *inserted_index = size() - 1;
+        return true;
+    }
+
+    template<typename C, typename U = T>
+    void insert_before_matching(U&& value, C callback, size_t first_index = 0, size_t* inserted_index = nullptr)
+    {
+        VERIFY(try_insert_before_matching(forward<U&&>(value), callback, first_index, inserted_index));
```

Comment:
Putting code inside `VERIFY(..)` makes me nervous, as someone will inevitably try to disable the VERIFY macro to do nothing for "performance" in the future.
We might save some headaches if we just don't put code we need to run in there?

Reply:
`VERIFY` will never be disabled. Instead we should add a new `ASSERT` for things that can be disabled. :)

That said, I agree that we should not do effectful things in `VERIFY` as it's much harder to read.

Reply:
What would be the preferred way to do it? I could redo them like:

```
auto did_allocate = try_insert_before_matching(forward<U&&>(value), callback, first_index, inserted_index);
VERIFY(did_allocate);
```

Reply:
Yes, that would be better.

Reply:
Alright, I've changed it to do that throughout.